internal/iam/query.go

```diff
@@ -309,6 +309,151 @@ const (
+                                      from user_group_roles)
+         and iam_grant.resource = any(@resources)
+    ),
+	global_role_special_grants as (
+      select iam_role_global.public_id             as role_id,
+             iam_role_global.scope_id              as role_scope_id,
+             'global'                              as role_parent_scope_id,
+             iam_role_global.grant_scope           as grant_scope,
+             roles_with_grants.canonical_grant     as canonical_grant
+        from iam_role_global
+        join roles_with_grants
+          on roles_with_grants.role_id = iam_role_global.public_id
+       where iam_role_global.grant_scope = any('{ children, descendants }')
+    ),
+    global_role_individual_grants as (
+      select iam_role_global.public_id             as role_id,
+             iam_role_global.scope_id              as role_scope_id,
+             'global'                              as role_parent_scope_id,
+             individual.scope_id                   as grant_scope,
+             roles_with_grants.canonical_grant     as canonical_grant
+        from iam_role_global
+        join roles_with_grants
+          on roles_with_grants.role_id = iam_role_global.public_id
+        join iam_role_global_individual_org_grant_scope individual
+          on individual.role_id = iam_role_global.public_id
+       where iam_role_global.grant_scope = 'individual'
+         and individual.scope_id = @request_scope
+    ),
+    org_roles as (
+      select iam_role_org.public_id             as role_id,
+             iam_role_org.scope_id              as role_scope_id,
+             'global'                           as role_parent_scope_id,
+             iam_role_org.scope_id              as grant_scope,
+             roles_with_grants.canonical_grant  as canonical_grant
+        from iam_role_org
+        join roles_with_grants
+          on roles_with_grants.role_id = iam_role_org.public_id
+       where iam_role_org.grant_this_role_scope
+         and iam_role_org.scope_id = @request_scope
+    ),
+    global_and_org_roles as (
+      select role_id,
+             role_scope_id,
+             role_parent_scope_id,
+             grant_scope,
+             canonical_grant
+        from global_role_special_grants
+       union
+      select role_id,
+             role_scope_id,
+             role_parent_scope_id,
+             grant_scope,
+             canonical_grant
+        from global_role_individual_grants
+       union
+      select role_id,
+             role_scope_id,
+             role_parent_scope_id,
+             grant_scope,
+             canonical_grant
+        from org_roles
+    )
+    select role_id,
+           role_scope_id,
+           role_parent_scope_id,
+           grant_scope,
+           canonical_grant as grant
+      from global_and_org_roles
+  group by role_id,
+           role_scope_id,
+           role_parent_scope_id,
+           grant_scope,
+		   canonical_grant;
```

Comment:
formatting: extra space 
```suggestion
	   canonical_grant;
```